pkg/apis/core/validation/shoot.go

```diff
@@ -556,6 +582,39 @@ func ValidateTotalNodeCountWithPodCIDR(shoot *core.Shoot) field.ErrorList {
 	return allErrs
 }
 
+// ValidateEncryptionConfigUpdate validates the updates to the KubeAPIServer encryption configuration.
+func ValidateEncryptionConfigUpdate(newConfig, oldConfig *core.EncryptionConfig, currentEncryptedResources sets.Set[string], etcdEncryptionKeyRotation *core.ETCDEncryptionKeyRotation, resourceType string, isClusterInHibernation bool, fldPath *field.Path) field.ErrorList {
+	var (
+		allErrs               = field.ErrorList{}
+		oldEncryptedResources = sets.New[string]()
+		newEncryptedResources = sets.New[string]()
+	)
+
+	if oldConfig != nil {
+		oldEncryptedResources.Insert(oldConfig.Resources...)
+	}
+
+	if newConfig != nil {
+		newEncryptedResources.Insert(newConfig.Resources...)
+	}
+
+	if !newEncryptedResources.Equal(oldEncryptedResources) {
+		if etcdEncryptionKeyRotation != nil && etcdEncryptionKeyRotation.Phase != core.RotationCompleted && etcdEncryptionKeyRotation.Phase != "" {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("resources"), fmt.Sprintf("resources cannot be changed when .status.credentials.rotation.etcdEncryptionKey.phase is not %q", string(core.RotationCompleted))))
+		}
+
+		if !oldEncryptedResources.Equal(currentEncryptedResources) {
+			allErrs = append(allErrs, field.Forbidden(fldPath.Child("resources"), fmt.Sprintf("resources cannot be changed when the %s resources in %s and the status.encryptedResources are not equal", resourceType, fldPath)))
```

Comment:
```suggestion
			allErrs = append(allErrs, field.Forbidden(fldPath.Child("resources"), "resources cannot be changed because a previous encryption configuration change is currently being rolled out")
```
This message could be a bit more intuitive for users.